src/citrine/resources/file_link.py

```diff
@@ -801,37 +832,67 @@ def file_processing_result(self, *,
 
         return results
 
-    def ingest(self, files: Iterable[FileLink]):
+    def ingest(self,
+               files: Iterable[FileLink],
+               *,
+               upload: bool = False,
+               raise_errors: bool = True,
+               build_table: bool = False,
+               delete_dataset_contents: bool = False,
+               delete_templates: bool = True,
+               ) -> "IngestionStatus":  # noqa: F821
         """
         [ALPHA] Ingest a set of CSVs and/or Excel Workbooks formatted per the gemd-ingest protocol.
 
         Parameters
         ----------
         files: List[FileLink]
-            A list of files, already on platform, from which GEMD objects should be built
+            A list of files from which GEMD objects should be built
+        upload: bool
+            If the files are off-platform references, upload them first.  Defaults to False,
+            in which case an off-platform resource raises an error.
+        raise_errors: bool
+            Whether ingestion errors raise exceptions (vs. simply reported in the results).
+            Default: True
+        build_table: bool
+            Whether to trigger a regeneration of the table config and building the table
+            after ingestion.  Default: False
+        delete_dataset_contents: bool
+            Whether to delete old objects prior to creating new ones.  Default: False
+        delete_templates: bool
+            Whether to delete old templates if deleting old objects.  Default: True
+
+        Returns
+        ----------
+        IngestionStatus
+            The result of the ingestion operation.
 
         """
+        from citrine.resources.ingestion import IngestionCollection  # noqa: F401
+
         targets = [self._resolve_file_link(f) for f in files]
-        if any(self._is_external_url(f.url) for f in targets):
-            externals = [f.url for f in targets if self._is_external_url(f.url)]
+        offplatform = [f for f in targets if not self._is_on_platform_url(f.url)]
+        if upload:
+            with TemporaryDirectory() as downloads:
+                paths = [Path(downloads) / f.filename for f in offplatform]
+                for file_link, path in zip(offplatform, paths):
+                    self.download(file_link=file_link, local_path=path)
+                onplatform = [self.upload(file_path=p) for p in paths]
```

Comment:
Would there be any value in handling failed "downloads" or uploads, to allow the rest of the files to be processed?

Reply:
For the current platform functionality, files need to be processed as a complete unit.  But there may be value in inserting a try-except here so that you find out all files that won't upload for whatever reason.  On the other hand, if I screwed up 20 paths, that'd flood a users screen without much actual improvement in signal.